File: samples/src/main/java/tk/zielony/carbonsamples/ComponentsActivity.java
package tk.zielony.carbonsamples;

import android.os.Bundle;

import java.io.Serializable;

import carbon.component.PaddingItem;
import tk.zielony.carbonsamples.component.AvatarTextListItemActivity;
import tk.zielony.carbonsamples.component.AvatarTextRatingSubtextDateListItemActivity;
import tk.zielony.carbonsamples.component.IconTextListItemActivity;
import tk.zielony.carbonsamples.component.ImageTextSubtextDateListItemActivity;
import tk.zielony.carbonsamples.component.RegisterActivity;

@ActivityAnnotation(title = R.string.componentsActivity_title)
public class ComponentsActivity extends SampleListActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Samples.initToolbar(this);

        setItems(new Serializable[]{
                new PaddingItem(getResources().getDimensionPixelSize(R.dimen.carbon_paddingHalf)),
                "Lists and forms composed of reusable components with data binding",
                new SampleActivityItem(IconTextListItemActivity.class),
                new SampleActivityItem(AvatarTextListItemActivity.class),
                new SampleActivityItem(ImageTextSubtextDateListItemActivity.class),
                new SampleActivityItem(AvatarTextRatingSubtextDateListItemActivity.class),
                new SampleActivityItem(RegisterActivity.class),
                new PaddingItem(getResources().getDimensionPixelSize(R.dimen.carbon_paddingHalf))
        });
    }

}

File: samples/src/main/java/tk/zielony/carbonsamples/ComponentsActivity.kt
package tk.zielony.carbonsamples

import android.os.Bundle
import carbon.component.PaddingItem
import tk.zielony.carbonsamples.component.*

@SampleAnnotation(
        titleId = R.string.componentsActivity_title,
        layoutId = R.layout.activity_samplelist,
        iconId = R.drawable.ic_view_compact_black_24dp
)
class ComponentsActivity : SampleListActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initToolbar()

        setItems(listOf(
                PaddingItem(resources.getDimensionPixelSize(R.dimen.carbon_paddingHalf)),
                "Lists and forms composed of reusable components with data binding",
                SampleActivityItem(IconTextListItemActivity::class.java),
                SampleActivityItem(AvatarTextListItemActivity::class.java),
                SampleActivityItem(ImageTextSubtextDateListItemActivity::class.java),
                SampleActivityItem(AvatarTextRatingSubtextDateListItemActivity::class.java),
                SampleActivityItem(RegisterActivity::class.java),
                SampleActivityItem(ComponentViewActivity::class.java),
                PaddingItem(resources.getDimensionPixelSize(R.dimen.carbon_paddingHalf))
        ))
    }

}
